public/model/geo/filter.ts

```diff
@@ -4,9 +4,22 @@
  */
 
 import { LatLon } from '@opensearch-project/opensearch/api/types';
-import { FilterMeta, GeoBoundingBoxFilter } from '../../../../../src/plugins/data/common';
+import { Polygon } from 'geojson';
+import {
+  Filter,
+  FilterMeta,
+  FILTERS,
+  GeoBoundingBoxFilter,
+} from '../../../../../src/plugins/data/common';
 import { GeoBounds } from '../map/boundary';
 
+export type FilterRelations = 'INTERSECTS' | 'DISJOINT' | 'WITHIN';
+
+export type GeoShapeFilter = Filter & {
+  meta: FilterMeta;
+  geo_shape: any;
```

Comment:
Could we define the type of geo_shape?

Reply:
Geo shape has variable field name property and fixed string property. How do you recommend create type in that case?

Reply:
Maybe we can define something like:

```
type GeoShape = {
  ignore_unmaped: Boolean,
  location: {
    relation: FilterRelations,
    shape: {
      type: 'Polygon',
      coordinates: [
        [
          [number, number],
          [number, number],
          [number, number],
          [number, number],
        ],
      ],
    },
  }
}
```

Reply:
but location is not fixed. It could be any field name

Reply:
In short term I think it's ok to use `any`, long term we need define the type to make it safer usage

Reply:
We might can use generic type for it?

```
type Coordinate = [number, number];

type Polygon = {
  type: 'Polygon';
  coordinates: Coordinate[][];
};

type Geometry = Polygon;

type Location<F extends string> = {
  [K in F]: {
    relation: 'INTERSECTS';
    shape: Geometry;
  };
};

type GeoShape<F extends string> = {
  ignore_unmaped: true;
} & Location<F> & Geometry;

```